system_setup/common/wsl_conf.py

```diff
@@ -0,0 +1,164 @@
+        },
+        "interop": {
+            "enabled": "true",
+            "appendwindowspath": "true"
+        }
+    },
+    "ubuntu": {
+        "GUI": {
+            "theme": "default",
+            "followwintheme": "false"
+        },
+        "Interop": {
+            "guiintegration": "false",
+            "audiointegration": "false",
+            "advancedipdetection": "false"
+        },
+        "Motd": {
+            "wslnewsenabled": "true"
+        }
+    }
+}
+
+
+def wsl_config_loader(data, pathname, config_ref, id):
+    """
+    Loads the configuration from the given file type,
+    section and reference config.
+
+    :param data: dict, the data to load into
+    :param pathname: string, the path to the file to load
+    :param id: string, the name of the section to load
+    """
+    if not os.path.exists(pathname):
+        return data
+    config = ConfigParser()
+    config.read(pathname)
+    for conf_sec in config:
+        if conf_sec in config_ref[id]:
+            conf_sec_list = config[conf_sec]
+            for conf_item in conf_sec_list:
+                if conf_item in config_ref[id][conf_sec]:
+                    data[conf_sec.lower()
+                         + "_" + conf_item.lower()] = \
+                             conf_sec_list[conf_item]
+    return data
+
+
+def default_loader(is_advanced):
+    """
+    This will load the default WSL config for the given type.
+
+    :param is_advanced: boolean, True if it is WSLConfigurationAdvanced,
+                        else is WSLConfigurationBase
+    """
+    data = {}
+    conf_ref = config_adv_default if is_advanced else config_base_default
+    data = wsl_config_loader(data, "/etc/wsl.conf", conf_ref, "wsl")
+    if is_advanced:
+        data = \
+            wsl_config_loader(data, "/etc/ubuntu-wsl.conf", conf_ref, "ubuntu")
+    return data
+
+
+# TODO: remove dryrun and add root param once we write the option to .subiquity
+def wsl_config_update(config_class, is_dry_run):
+    """
+    This update the configuration file for the given class.zzd
+
+    :param config_class: WSLConfigurationBase or WSLConfigurationAdvanced
+    :param is_dry_run: boolean, True if it is a dry run
+    """
+    if is_dry_run:
+        log.debug("mimicking setting config %s", config_class)
+    temp_conf_default = {}
+    temp_confname = config_class.__str__()
+    if temp_confname.startswith("WSLConfigurationBase"):
+        temp_conf_default = config_base_default
+    elif temp_confname.startswith("WSLConfigurationAdvanced"):
+        temp_conf_default = config_adv_default
+    else:
+        raise TypeError("Invalid type name.")
+    if is_dry_run:
+        return
+
+    # update the config file
+    for config_type in temp_conf_default:
+        config_sections = temp_conf_default[config_type]
+
+        config = ConfigParser()
+        config.BasicInterpolcation = None
+
+        if config_type == "wsl":
+            conf_file = "/etc/wsl.conf"
```

Comment:
There is some repetiution on the file path now between `default_loader` and `wsl_config_update`. Put it as a constant in the file.

Reply:
done